plugins/tabular_dataset.cc

```diff
@@ -106,6 +109,83 @@ struct TabularDataset::TabularDataStore: public ColumnIndex, public MatrixView {
         return result;
     }
 
+    virtual std::vector<CellValue>
+    getColumnDense(const ColumnName & column) const
+    {
+        auto it = columnIndex.find(column);
+        if (it == columnIndex.end()) {
+            throw HttpReturnException(400, "Tabular dataset contains no column with given hash",
+                                      "columnHash", column,
+                                      "knownColumns", columnNames);
+        }
+
+        std::vector<CellValue> result;
+        result.reserve(rowCount);
+
+        for (unsigned i = 0;  i < chunks.size();  ++i) {
+            auto onValue = [&] (size_t n, CellValue val)
+                {
+                    result.emplace_back(std::move(val));
+                    return true;
+                };
+            
+            chunks[i].columns[it->second].forEach(onValue);
+        }
+        
+        return result;
+    }
+
+    virtual std::tuple<BucketList, BucketDescriptions>
+    getColumnBuckets(const ColumnName & column, int maxNumBuckets) const override
+    {
+        auto it = columnIndex.find(column);
+        if (it == columnIndex.end()) {
+            throw HttpReturnException(400, "Tabular dataset contains no column with given hash",
+                                      "columnHash", column,
+                                      "knownColumns", columnNames);
+        }
+
+        std::unordered_map<CellValue, size_t> values;
+        std::vector<CellValue> valueList;
+
+        size_t totalRows = 0;
+
+        for (unsigned i = 0;  i < chunks.size();  ++i) {
+            auto onValue = [&] (CellValue val, size_t /* count */)
+                {
+                    if (values.insert({val,0}).second)
+                        valueList.push_back(std::move(val));
+                    return true;
+                };
+
+            chunks[i].columns[it->second].forEachDistinctValue(onValue);
```

Comment:
This is somewhat strange that you have to use use a map to filter duplicates when calling ```forEachDistinctValue```.  Is the name of the function misleading?